get-song-name.js

```diff
@@ -1,56 +1,161 @@
+        const artistName = nowPlaying.querySelector('div.artist span').textContent;
+        scrapedSong.artist = artistName;
+        scrapedSong.albumName = album;
+      }
+      const infoLink = 
+        isPlaying.querySelector('div.info a') ||
+        isPlaying.querySelector('div.collection-title-details a.item-link');
+      if(infoLink) {
+        const albumURL = infoLink.href;
+        scrapedSong.albumURL = albumURL;
+      }
+      var progressTitle = carouselPlayer.querySelector('.progress-transport .title');
+      if(progressTitle) {
+        const songName = progressTitle.querySelector('span[data-bind="text: currentTrack().trackTitle"]').textContent;
+        const songNumber = progressTitle.querySelector('span[data-bind="text: currentTrack().trackNumber"]').textContent;
+        scrapedSong.songNumber = songNumber;
+        scrapedSong.songName = songName;
       }
     }
+  })
+}
 
-    if (artistName) {
-      scrapedSong.artistName = artistName;
-    }
-  });
+// Bandcamp featured page
+// examples: https://daily.bandcamp.com/features 
+var actualPlayer = document.querySelector('mplayer.playing');
+if (actualPlayer) {
+  const albumURL = actualPlayer.querySelector('a.mptralbum').href;
+  const artistURL = actualPlayer.querySelector('a.mpartist').href;
+  const album = actualPlayer.querySelector('a.mptralbum').textContent;
+  const artist = actualPlayer.querySelector('a.mpartist').textContent;
+  var songName = actualPlayer.querySelector('.mptracktitle').textContent;
+  const songNumber = actualPlayer.querySelector('.mptracknumber').textContent;
+  const songURL = artistURL+"track/"+songName.replace(' ','-')
+  scrapedSong.artistURL = artistURL;
+  scrapedSong.albumURL = albumURL;
+  scrapedSong.albumName = album; 
+  scrapedSong.artist = artist;
+  scrapedSong.songName = songName;
+  scrapedSong.songNumber = songNumber;
+  scrapedSong.songURL = songURL;
+}
 
-  
-  const albumSong = document.querySelector('.track_row_view');
-  
+// Bandcamp  weekly player
+// https://bandcamp.com/?show=47
+var bandcampPlayer = document.querySelector('.bcweekly-info .row .bcweekly-info-inner .bcweekly-tracks .bcweekly-current');
+if(bandcampPlayer) {
+  const artist = bandcampPlayer.querySelector('p.track-artist a').textContent;
+  const artistURL = bandcampPlayer.querySelector('p.track-artist a').href;
+  var trackDetails = bandcampPlayer.querySelector('.track-details');
+  const albumURL = trackDetails.querySelector('a').href;
+  const songName = trackDetails.querySelector('a span.track-title').textContent;
+  const album = trackDetails.querySelector('a span.track-album').textContent;
+  scrapedSong.artist = artist;
+  scrapedSong.artistURL = artistURL;
+  scrapedSong.albumName = album;
+  scrapedSong.albumURL = albumURL;
+  scrapedSong.songName = songName;
+
+}
+
+// Bandcamp hompage discover section
+// https://bandcamp.com/
+var discover = document.querySelector('.discover-item .playing');
```

Comment:
I think as a pattern we want to stay away from var and use let or const instead. var would be for strictly needing hoisting.

Reply:
Resolved in commit 32dc34d